covsirphy/cleaning/jhu_complement.py

```diff
@@ -230,7 +271,9 @@ def _recovered_partial(self, df):
         max_frequency = series.value_counts().max()
         if max_frequency <= self.interval:
             return df
-        # Complement
+        # Complement first any ending unupdated values
+        df = self._recovered_partial_ending(df)
```

Comment:
Can we move this line to `_protocol` method?

Reply:
I would say no, because the user needs to know that partial complement in general was applied. It is too specific to inform him about in-between and ending values, there is no need to do that. Both are partial.

Reply:
I mean `(self.R, self._recovered_partial_ending, 3)`.
3 is the point of this method and 3 means "partially" as defined in `cls.STATUS_NAME_DICT`.

Reply:
Just to confirm, can we apply `df = self._recovered_partial_ending(df)` when NOT `max_frequency <= self.interval`?

Reply:
Oh, yes then it could be added there, right before partial line inside `_protocol`.

Reply:
Thank you!

Reply:
>Just to confirm, can we apply df = self._recovered_partial_ending(df) when max_frequency <= self.interval?
>
There is no point in doing that, since the ending values will be unique and not duplicates. Either way this will be captured by `sel_0 = (df[self.R] == r_max).sum() > self.max_ending_unupdated`